list-navigation/lambda/session-state.ts

```diff
@@ -48,14 +49,19 @@ export class ListNavSessionState {
     // instead of the page token passed via API arguments
     upcomingPageToken?: PageToken;
 
+
+    //Required for pagination in case of DDBListProvider, used by record-event APIs
+    pageStack?: PageToken[];
```

Comment:
why are we modifying the common session state to store this just a for a single provider? if the provider needs to store this itself to satisfy the provider interface then that is fine, but we shouldn't be modifying code outside the provider for the benefit of a single provider.